assets/src/blocks/ENForm/ENFormFrontend.js

```diff
@@ -0,0 +1,487 @@
+    text_below_button,
+    button_text,
+  } = attributes;
+
+  const is_side_style = en_form_style === 'side-style';
+  // Keep extra content between repaints
+  const extra_content = document.querySelector('.enform-extra-header-placeholder')?.innerHTML;
+
+  return (
+    <div className="enform" id="enform">
+      <div id="enform-content">
+
+        <div className="title-and-description">
+          {title &&
+            <h2 dangerouslySetInnerHTML={{__html: title ? unescape(title) : ''}} />
+          }
+          {is_side_style &&
+            <div className="enform-extra-header-placeholder"
+              dangerouslySetInnerHTML={{__html: extra_content ? unescape(extra_content) : ''}} />
+          }
+          {description &&
+            <div className="form-description" dangerouslySetInnerHTML={{__html: unescape(description)}} />
+          }
+        </div>
+
+        <div className="form-container">
+          <form
+            id="p4en_form"
+            name="p4en_form"
+            method="post"
+            noValidate
+            onSubmit={onFormSubmit}
+          >
+            <div className={en_form_style === 'full-width-bg' ? 'row' : ''}>
+              <div className={en_form_style === 'full-width-bg' ? 'col-md-8' : ''}>
+                <FormGenerator {...{fields, attributes, onInputChange, onBlur, errors}} />
+              </div>
+
+              <div className={en_form_style === 'full-width-bg' ? 'col-md-4 submit' : 'submit'}>
+                <button type="submit" form="p4en_form" name="p4en_form_save_button" id="p4en_form_save_button" className="btn btn-primary" >
+                  { button_text ? unescape(button_text) : __('Sign', 'planet4-engagingnetworks') }
+                </button>
+                <div className="enform-notice"></div>
+                {en_form_style === 'full-width-bg' &&
+                  <div className="enform-legal">
+                    <p dangerouslySetInnerHTML={{__html: text_below_button ? unescape(text_below_button) : ''}} />
+                  </div>
+                }
+              </div>
+
+              {en_form_style !== 'full-width-bg' &&
+                <div className="enform-legal">
+                  <p dangerouslySetInnerHTML={{__html: text_below_button ? unescape(text_below_button) : ''}} />
+                </div>
+              }
+            </div>
+            {error_msg &&
+              <span className="enform-error">{ error_msg }</span>
+            }
+          </form>
+          <div id="form-data" data-postdata={JSON.stringify(makePostData(form_data, fields))} />
+        </div>
+      </div>
+    </div>
+  );
+};
+
+const submitENForm = props => {
+  const {
+    form_data,
+    fields,
+    enform_goal,
+    thankyou_url,
+    setErrorMsg,
+    setActiveTplId,
+    en_page_id,
+  } = props;
+
+  const post_data = makePostData(form_data, fields);
+
+  // Send form
+  const post_url = `${window.p4bk_vars.siteUrl}/wp-json/planet4/v1/enform/${en_page_id}`;
```

Comment:
This endpoint need to move in master theme.
https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/main/classes/rest/class-rest-api.php#L376-L390

Reply:
At some point we have to move the `p4bk_vars` to master-theme as well. Could you please check if it is present in blocks plugin and master theme, does that make any issue? 🤔

Reply:
[Updated](https://github.com/greenpeace/planet4-master-theme/pull/2190/commits/b95367b2004b734d88ecfd36abb9e0398430017e)! I've moved `themeUrl` to the `p4_vars` we use in the master theme for now. For the siteUrl we can use the dataset NRO value.